service/featuregate/gates.go

```diff
@@ -27,38 +27,45 @@ type Gate struct {
 	Enabled     bool
 }
 
-var reg = &registry{gates: make(map[string]Gate)}
+var reg = &Registry{gates: make(map[string]Gate)}
 
 // IsEnabled returns true if a registered feature gate is enabled and false otherwise.
 func IsEnabled(id string) bool {
-	return reg.isEnabled(id)
+	return reg.IsEnabled(id)
 }
 
 // List returns a slice of copies of all registered Gates.
 func List() []Gate {
-	return reg.list()
+	return reg.List()
 }
 
 // Register a Gate. May only be called in an init() function.
 // Will panic() if a Gate with the same ID is already registered.
 func Register(g Gate) {
-	if err := reg.add(g); err != nil {
+	if err := reg.Add(g); err != nil {
 		panic(err)
 	}
 }
 
 // Apply a configuration in the form of a map of Gate identifiers to boolean values.
 // Sets only those values provided in the map, other gate values are not changed.
 func Apply(cfg map[string]bool) {
-	reg.apply(cfg)
+	reg.Apply(cfg)
 }
 
-type registry struct {
+// GlobalRegistry returns the Registry instance used by package-global functions in this package.
+func GlobalRegistry() *Registry {
+	return reg
+}
+
+type Registry struct {
```

Comment:
What is the need in this PR that adds the flag to make this public?

Reply:
You still did not answer why is this needed.

Reply:
I understand where this comes from, but my comment was to remove the other global funcs.